kubernetes-1.31.yaml

```diff
@@ -161,17 +163,26 @@ subpackages:
 
           install -d ${{targets.subpkgdir}}/var/log/kube-apiserver
 
-  - name: kubernetes-pause-3.9
+  - name: kubernetes-pause-${{vars.kubernetes-version}}
```

Comment:
Why do we put the pause version in the package name instead of following the kubernetes version? It's built out of the same repository, and we aren't really doing anything too special with it

Reply:
thanks for the review.
we were using the naming because pause comes from k/k repository but follows a different versioning pattern. e.g. all components of k/k gets releases under 1.31.x as of now but pause version is 3.10. It's basically a single pause.c file. 

In this patch, I moved from having `3.9` in the name to appending `-${{vars.kubernetes-version}}` because it was coming from the same repository and given we are always building the pause binary from source we can trust this. As per the docs [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/updating-pause-images.md#first-level-core-projects), pause gets updated first in k/k repo and then ecosystem follows so we can always trust this one.

Reply:
It is wrong we put the version in the package name here regardless though. Especially when it's primarily only used to tag the image. We should be getting that with the image module and not relying on an improper package name to decipher it. This will lead to drift when pause updates

Look at how we handle this with EKS Distro. None of those components diverge from the EKS and kubernetes version at all in spite of having different internal versions. We can do terraform magic instead if we need this for the tag but we shouldn't be doing it at the package level

Reply:
we are now using the name which matches the kubernetes version `v1.31.x` so no more dealing with 3.9 and 3.10 because we are always building it from source and given we are grepping the TAG from Makefile, the version is also set correctly. 

It also matches the convention that we are following with other packages _(kube-proxy-1.31)_. Now, the packages are named something like `kubernetes-pause-1.31`, `kubernetes-pause-1.30` etc.